kube_janitor/janitor.py

```diff
@@ -50,11 +51,24 @@ def matches_resource_filter(
     )
 
 
-def get_ttl_expiry_time(resource, ttl_seconds: int) -> datetime.datetime:
-    creation_time = datetime.datetime.strptime(
-        resource.metadata["creationTimestamp"], "%Y-%m-%dT%H:%M:%SZ"
+def get_deployment_time(
+    resource, deployment_time_annotation: Optional[str]
+) -> datetime.datetime:
+    metadata = resource.metadata
+    annotations = metadata.get("annotations", {})
+
+    creation_timestamp = metadata["creationTimestamp"]
+    deployment_timestamp = (
+        annotations.get(deployment_time_annotation)
+        if deployment_time_annotation
+        else None
+    )
+
+    return max(
+        datetime.datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%SZ")
```

Comment:
I think this should handle the case where the annotation format is wrong --> I would ignore the annotation and use creation time in this case (and log warning)

Reply:
Good point, fixed.